app/instrumentation-tests/src/org/commcare/androidTests/DateWidgetsTests.kt

```diff
@@ -0,0 +1,208 @@
+import androidx.test.espresso.matcher.ViewMatchers.*
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.filters.LargeTest
+
+import org.commcare.annotations.BrowserstackTests
+import org.commcare.dalvik.R
+import org.commcare.utils.InstrumentationUtility
+import org.commcare.utils.isPresent
+import org.hamcrest.Matchers
+import org.hamcrest.Matchers.allOf
+
+import org.junit.Assert.*
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import java.text.SimpleDateFormat
+import java.time.LocalDate
+import java.time.format.DateTimeFormatter
+import java.util.*
+import kotlin.math.absoluteValue
+
+
+
+@RunWith(AndroidJUnit4::class)
+@LargeTest
+@BrowserstackTests
+class DateWidgetsTests: BaseTest() {
+    companion object {
+        const val CCZ_NAME = "date_widgets_tests.ccz"
+        const val APP_NAME = "Date Widgets"
+        val listOfNepaliMonths = listOf("Baishakh","Jestha","Ashadh","Shrawan",
+                            "Bhadra","Ashwin","Kartik","Mangsir","Poush","Magh",
+                            "Falgun","Chaitra")
+        val listOfEthiopianMonths = listOf("Säne","Hämle","Nähäse","P’agume",
+            "Mäskäräm","T’ïk’ïmt","Hïdar","Tahsas","T’ïr","Yäkatit",
+            "Mägabit","Miyaziya")
+
+    }
+
+    @Before
+    fun setup() {
+        installApp(APP_NAME, CCZ_NAME)
+        InstrumentationUtility.login("test1", "123")
+    }
+
+    /**
+     * Adding the teardown method so the tests doesnot fail after only thefirst execution, when executed as a whole.
+     */
+
+    @Test
+    fun testNepalDateWidgets(){
+
+        InstrumentationUtility.openModule(0)
+        onView(withSubstring("This form will test the Nepal date widget.")).isPresent()
+        InstrumentationUtility.nextPage()
+        InstrumentationUtility.nextPage()
+
+        for (i in 1..12){
+            val month_text = InstrumentationUtility.getText(onView(withId(R.id.monthtxt)))
+            if (month_text != null) {
+                assertTrue(listOfNepaliMonths.contains(month_text))
+            }
+            onView(withId(R.id.monthupbtn)).perform(ViewActions.click())
+        }
+        onView(withId(R.id.yeardownbtn)).perform(ViewActions.click())
+        val date_selected = setDateToUniversalCalender(-10,0,0)
+        val gregorian_date = InstrumentationUtility.getText(onView(withId(R.id.dateGregorian)))
+        val formatted_date = formatGregorianDate(gregorian_date.drop(1).dropLast(1))
+
+        InstrumentationUtility.nextPage()
+        Log.i("Nepali date:", date_selected)
+        Log.i("Gregorian date:", formatted_date)
+        assertTrue(onView(withSubstring(formatted_date)).isPresent())
+        assertTrue(onView(withSubstring(date_selected)).isPresent())
+        InstrumentationUtility.submitForm()
+        assertTrue(onView(ViewMatchers.withText("1 form sent to server!")).isPresent())
+    }
+
+
+    @Test
+    fun testEthiopianDateWidgets(){
```

Comment:
this test looks exactly simiar to `testNepalDateWidgets` except that you are passing a different set of months. You can create a function - `testDateWidgetWithMonths(listOfMonths : List)` and put all your test code inside that function. And then just call this function with the corresponding months list in the individual tests.

Reply:
merged both testcases into 1 and looped through the listOflists